images/rootfs.yml.in

```diff
@@ -28,26 +28,37 @@ onboot:
 services:
    - name: rsyslogd
      image: RSYSLOGD_TAG
+     cgroupsPath: /eve/eve-rsyslogd
    - name: ntpd
      image: linuxkit/openntpd:v0.5
+     cgroupsPath: /eve/eve-ntpd
    - name: sshd
      image: linuxkit/sshd:v0.5
+     cgroupsPath: /eve/eve-sshd
    - name: wwan
      image: WWAN_TAG
+     cgroupsPath: /eve/eve-wwan
    - name: wlan
      image: WLAN_TAG
+     cgroupsPath: /eve/eve-wlan
    - name: lisp
      image: LISP_TAG
+     cgroupsPath: /eve/eve-lisp
    - name: guacd
      image: GUACD_TAG
+     cgroupsPath: /eve/eve-guacd
    - name: pillar
      image: PILLAR_TAG
+     cgroupsPath: /eve/eve-pillar
    - name: vtpm
      image: VTPM_TAG
+     cgroupsPath: /eve/eve-vtpm
    - name: watchdog
      image: WATCHDOG_TAG
+     cgroupsPath: /eve/eve-watchdog
    - name: xen-tools
      image: XENTOOLS_TAG
+     #Since all the AppInstances will be child process of xen-tools we will not restrict xen-tools under eve cgroup.
```

Comment:
I don't think this is correct since we create brand new tasks and not children of xen-tools. Can you please verify that adding xen-tools to the same list will break anything?

Reply:
When `xen-tools` was added as part of `eve` cgroup, deploying an AppInstance triggered oom_reaper which (sometimes) killed an EVE service leading to EVE crash. 

```
(ns: pillar) 63434710-069c-4648-b0eb-6ae94b8d6cb5:/# ps aux | grep qemu
 7535 root      4:21 /usr/lib/xen/bin/qemu-system-x86_64 -display none -S -no-user-config -nodefaults -no-shutdown -overcommit mem-lock=on -overcommit cpu-pm=on -serial chardev:charserial0 -no-hpet -name ubuntu-bionic-18.04.2 -readconfig /var/run/domainmgr/xen/xen2.cfg -pidfile /var/run/hypervisor/kvm/ubuntu-bionic-18.04.2/pid
 9186 root      0:00 grep qemu
(ns: pillar) 63434710-069c-4648-b0eb-6ae94b8d6cb5:/# grep -r 7535 /hostfs/sys/fs/cgroup/memory/*
/hostfs/sys/fs/cgroup/memory/eve/eve-xen-tools/cgroup.procs:7535
/hostfs/sys/fs/cgroup/memory/eve/eve-xen-tools/tasks:7535
(ns: pillar) 63434710-069c-4648-b0eb-6ae94b8d6cb5:/#
```

In the above example, I've included `xen-tools` in `eve` cgroup. I deployed `ubuntu-bionic-18.04.2` app and noticed the app's pid was added as part of `/hostfs/sys/fs/cgroup/memory/eve/eve-xen-tools/cgroup.procs`. 
Since `xen-tools` was under `eve` cgroup, the limit mentioned for `eve` got applied to the appInstance as well.

Reply:
When you do `xl unpause`, how is the process created? If it is a child of anything in `xen-tools`, then it will apply. I had thought it was separate, but maybe it is not?

If it is, then I would propose that we create a new cgroup, perhaps `eve-user-apps`, with no limits, and have a child cgroup for each app.

Mind you, if we use containerd and runc for this, we should inherit cgroups. But that is a bit off.

Reply:
OK, let me start by answering @deitch questions first:
> When you do xl unpause, how is the process created?

The process is already created. The creation of the process is a very traditional containerd dance of simply creating a task in a brand new container. That brand new container has nothing to do with xen-tools. In fact, it even gets created in a separate namespace (eve-user-apps).

>  If it is a child of anything in xen-tools, then it will apply. I had thought it was separate, but maybe it is not?

It is NOT a child for sure.

Now @adarsh-zededa -- we really need to look into this -- since it may single some kind of a bug/misconfiguration in how we're using containerd task creation mechanisms. So I still don't quite feel that we've answered all the questions here.

Reply:
> It is NOT a child for sure.

Then it definitely should not inherit the cgroup limits of `xen-tools`. Thanks for clarifying.

Reply:
Fixed this. Earlier in `LKTaskPrepare()` the new container inherited the cgroupPath from `xen-tools` config. 
Now all AppInstances container will have their own cgroup under a parent cgroup `eve-user-apps`.